src/main/java/io/quarkus/gizmo2/impl/Neg.java

```diff
@@ -4,13 +4,19 @@
 import java.util.function.BiFunction;
 
 import io.github.dmlloyd.classfile.CodeBuilder;
+import io.github.dmlloyd.classfile.TypeKind;
 import io.quarkus.gizmo2.Expr;
 
+import static io.smallrye.common.constraint.Assert.impossibleSwitchCase;
+
 final class Neg extends Item {
     private final Item a;
 
     Neg(final Expr a) {
         this.a = (Item) a;
+        if (a.typeKind() == TypeKind.REFERENCE || a.typeKind() == TypeKind.VOID) {
```

Comment:
Maybe add `BOOLEAN` in there too?

Reply:
`BOOLEAN` is an `INT` and I've accidentally verified before that `logicalNot(booleanExpr)` works (even if it generates strange bytecode), so I don't think we strictly _need_ it. It's probably a good idea though.

Reply:
Sorry, my bad, `logicalNot()` doesn't translate to `Neg`. Will add a `BOOLEAN` case.

Reply:
Done.